### ISSUE TITLE
Phsp element

### ISSUE THREAD

**Issue by weidenka** [ComPWA/ComPWA#259]
ComPWA currently does not properly include a phsp element in the calculation of the intensity. I would suggest do extend the XML model:
```xml
<Intensity Class="IncoherentIntensity" Name="4body">
  <!-- Add phsp element
         dPhi = |p_012*| |p3| dm_012 dOmega*_1 dOmega_3
         Final states (012) versus (3) are the default values if none
         are specified. This should be fine for most cases. -->
  <PhspElement FinalState1="0 1 2" FinalState2="3"/>
  <Intensity Class="StrengthIntensity" Name="4body">
    ....
  </Intensity>
</Intensity>
```
For the calculation of the phsp element the invariant mass m_012 and the masses of the final state particles are needed (see [Kinematics review](http://pdg.lbl.gov/2019/reviews/rpp2018-rev-kinematics.pdf)) The `IntensityBuild` would then add the subsystem 012 to the `Kinematics` class and create a FunctionTree that calculates the  `|p_1^*| \times |p_3|`. The functionallity to create this tag should be added to the expertsystem as well.

**Comment by redeboer** [comment]
Can't this information be inferred from the kinematics section in the XML? (If so, do we want the intensity section to be independent of the kinematics section?)

**Comment by weidenka** [comment]
Good point.

**Comment by spflueger** [comment]
Yeah, it already exists in the kinematics section. However its currently not used in the FunctionTree.

**Comment by spflueger** [comment]
So if I understand correctly, it would be enough to put these phase space volume factors into the phase space integrals of the normalization. Atm that line was commented out, so I would just put that back in and that should solve this issue. Right?

**Comment by weidenka** [comment]
I'm not sure if I understand what you mean. But the PHSP elements are calculated on a per-event basis. It is not the PHSP volume (which was implemented by Remco recently). I think we should add the calculation to `convertEventsToDataSet()` such that another `MDouble` is returned which contains the PHSP elements. But I agree that no special section is needed in the XML model.

**Comment by spflueger** [comment]
Ups. I just noticed I misread this. I thought we are talking about the phsp volume. Sorry about this. The phsp volume part I just fixed now. Although some code should be added to the constructor of the `HelicityKinematics`, which would call the functions determining the phase space integral. (Otherwise the value is just 1.0, unless stated in the XML).

Regarding this issue: In my opinion the phsp element should be calculated inside the Kinematics class, strictly from the knowledge of the final state.

**Comment by spflueger** [comment]
> I think we should add the calculation to convertEventsToDataSet() such that another MDouble is returned which contains the PHSP elements. But I agree that no special section is needed in the XML model.

That sounds good. Btw, I believe that these breakup momenta are all we need to calculate and multiply our Amplitudes with. So for each two body decay the corresponding break up momentum is multiplied with the decay amplitude, thats it! So I would calculate `M, theta, phi, phsp_element` for each two body decay and save this in the DataSample. It would be good to accumulate all of the part phsp elements into the whole one of the event (to save memory and gain speed)

**Comment by weidenka** [comment]
As far as I understand the PDG review. One needs to calculate it only for one subsystem.

**Comment by spflueger** [comment]
I'm not sure I understand you correctly. What do you mean with "calculate it only for one subsystem"? I assume you mean for only one topology of the full N body decay.

That make sense to me. I would assume that the total phsp element factors would be identical for every decay topology.

I think we just have to be careful when coming up with the implementation design. If my understanding of the whole problem is correct, simply calculating them for each two body decay and propagating them through the decay tree would be a general solution and leaves little room for implementation errors.

I suggest we come up with a concrete and full solution to this, and then discuss it and possibly improve it in more detail. Anyone wants to tackle this problem?

**Comment by spflueger** [comment]
So in my comparison with other frameworks. I don't not see them including any phsp element factors. I think we have to figure out if the Intensity we are calculating actually corresponds to the matrix element `M` or to the full decay rate Gamma. According to [this](http://pdg.lbl.gov/2012/reviews/rpp2012-rev-dalitz-analysis-formalism.pdf) it should be M, which means that a lot of other frameworks are also missing this factor. That would be really disturbing ...

**Comment by weidenka** [comment]
I agree that we actually calculate M and not the decay rate (which we should calculate). Other frameworks might include the additional factor at other places. Nevertheless, our implementation is currently not correct.

**Comment by spflueger** [comment]
> Nevertheless, our implementation is currently not correct.

Currently I believe that our current implementation is correct (and also all other frameworks. This is based on the fact that our kinematic distribution matches other frameworks perfectly). What I mentioned above was not very clear and correct.
The important part is that transition rate `Gamma` and the phase space element are defined in the same coordinate system as the matrix element `M`. In the mentioned [link](http://pdg.lbl.gov/2012/reviews/rpp2012-rev-dalitz-analysis-formalism.pdf), this is actually the case since the phase space element are just the invariant masses of subsystems used in the Dalitz plot and the BreitWigner.
